.github/workflows/sync_submodules.yml

```diff
@@ -45,28 +47,24 @@ jobs:
           git submodule update --init --recursive
           git submodule update --remote --recursive
 
-      - name: Check if changes were made and commit if found
-        id: check_and_commit_changes
+      - name: Check if changes were made and create pull request if found
+        id: check_and_create_pull_request
         run: |
-          if [ -z "$(git status --porcelain)" ]; then
-            echo "No changes were made."
```

Comment:
Probably we would want to check if there are changes to commit before trying to open a pull request and exit early if not. Otherwise a branch with no changes and a pull request for that branch will be created.

Reply:
Okay! Should we reflect that on the other workflows that create pull-requests?

Reply:
The other ones check both beforehand, whether there are changes and whether or not a branch already exists.